### PR TITLE
feat: add (in)compatibility runner/builder test

### PR DESCRIPTION
**Pull request by hacdias**
This tests the compatibility and incompatibility between builder and runner through a simple E2E test where we spin a server and then make requests to it.

For this, I created a placebo test plan which does nothing and it might be useful for future testing.

License: MIT
Signed-off-by: Henrique Dias <hacdias@gmail.com>

### CODE REVIEW

File: plans/do-nothing/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Do Nothing
+
+This is a placebo test for testing purposes.
```
Comment (daviddias) [inline plans/do-nothing/README.md]: oh, call the test placebo then :D

Comment (hacdias) [reply plans/do-nothing/README.md]: Done!

File: pkg/server/server_test.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+		t.Fail()
+	}
+}
+
+func TestCompatibleBuilder(t *testing.T) {
+	api := client.New(getServer())
+
+	req := &client.BuildRequest{
+		Plan:    "placebo",
+		Builder: "exec:go",
```
Comment (daviddias) [inline pkg/server/server_test.go]: No runner?

Comment (hacdias) [reply pkg/server/server_test.go]: This is the build request so yes, no runner.

Comment (daviddias) [reply pkg/server/server_test.go]: So you are building the placebo to then use it with the simple-add runner?

Why call this test `TestCompatibleBuilder` then? Shouldn't it be TestINCompatibleBuilder?

File: pkg/server/server_test.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+	"testing"
+
+	"github.com/ipfs/testground/pkg/daemon/client"
+)
+
+var srv *Server
```
Comment (nonsense) [inline pkg/server/server_test.go]: I think it'd be better to not have global vars in tests. We want to be able to run parallel tests. Also the port of the server should be given by the kernel, so just us ":0", and not a hardcoded one.

File: pkg/server/server_test.go (hunk 2 above)
Comment (hacdias) [reply pkg/server/server_test.go]: This test tests builder `exec:go` (line 87, 108) with runner `local:exec` so everything is correct.

File: pkg/server/server_test.go (hunk 3 above)
Comment (hacdias) [reply pkg/server/server_test.go]: Doooone!

File: pkg/server/server_test.go (hunk 2 above)
Comment (daviddias) [reply pkg/server/server_test.go]: image

**Pull request merged**